pkg/kv/kv.go

```diff
@@ -384,6 +385,21 @@ func (kvcodec *tableKVEncoder) AddRecord(
 	return Pairs(pairs), size, nil
 }
 
+// get record value for auto-increment field
+//
+// See: https://github.com/pingcap/tidb/blob/47f0f15b14ed54fc2222f3e304e29df7b05e6805/executor/insert_common.go#L781-L852
+// TODO: merge this with pkg/lightning/backend/kv/sql2kv.go
+func getAutoRecordID(d types.Datum, target *types.FieldType) int64 {
```

Comment:
I think we should make the function `getAutoRecordID` in the TiDB repo be a public interface to avoid further inconsistent behaviors.